.github/actions/build-engine/action.yaml

```diff
@@ -0,0 +1,109 @@
+# .github/actions/build-engine/action.yml
+name: ""
```

Comment:
Missing the name. Also, as a matter of completeness, should this be called `build-and-upload-engine`? Should it actually be 2 actions?